### DISCUSSION TITLE
Distribution for iOS when using shorebird is not executed

### DISCUSSION THREAD

**Discussion by immadisairaj** [codemagic-ci-cd/codemagic-docs#2564] [Q&A] [answered]
I was trying to build iOS app with shorebird and distribute app to TestFlight.

Due to some issue with shorebird signing, I was using custom signing to build (here it uses the codemagic and my signing).

The issue here is, after the build has happened it says the build has passed where I was not able to see the build on AppStore Connect.

Shorebird has a limitation on when uploading to AppStore Connect which is unchecking the "Manage version" (something).

And I need the build to be able to go to external test group.

I think this might be a limitation from shorebird? Or I am doing it wrong? I am not sure.

Build ID: 66571f5e5cf3872cfe89d3e5

App Repository: https://github.com/immadisairaj/radiosai

Shorebird iOS release docs: https://docs.shorebird.dev/guides/release/ios/

**Comment by himesh-cm** [comment]
Hey @immadisairaj, I am investigating the issue and will keep you posted.

**Reply by himesh-cm** [reply]
Hey @immadisairaj,

I noticed that you are trying to build a no-codesign build by running `shorebird release ios—-no-codesign`.
To run a codesign build

Please add -`-export-options-plist=/Users/builder/export_options.plist` to your shorebird release command and rerun the build.

Let me know how it goes.

**Reply by immadisairaj** [reply]
Thanks for looking into this @himesh-cm 

It did not work. This is the build ID of my try `6675925e8d9f918904ebedc8`

It gives me an error of
```
Export options plist /Users/builder/export_options.plist does not set manageAppVersionAndBuildNumber to false. This is required for shorebird to work.
```

Is there any way I could disable that flag when generating the export options?

**Reply by himesh-cm** [reply, marked as answer]
Hey @immadisairaj 

With the latest version of shorebird cli, it seems you have to set ​manageAppVersionAndBuildNumber to false in `/Users/builder/export_options.plist.`

You can use PlistBuddy to do so, which you can run after the Apply Profiles step
​
`​/usr/libexec/PlistBuddy -c 'Add :manageAppVersionAndBuildNumber bool false' /Users/builder/export_options.plist`

**Reply by immadisairaj** [reply]
I have tried that @himesh-cm 

But, have come back to the same problem at the publishing step. From what I am seeing, it is failing to get the artifacts (or maybe shorebird is not generating `*.ipa` artefacts?).

Build Id: `667933688d9f918904f1dd00`

**Reply by himesh-cm** [reply]
1. Please remove `--no-codesign`, this flag is not required while generating the IPA file.
2. Setup local properties is required for Android builds, so you may remove this as well.
3. You have not uploaded profiles for bundle id `com.immadisairaj.saivoice`, please add the profile from your apple developer account, upload them on codemagic > team settings and re-run the build on codemagic

**Reply by immadisairaj** [reply]
Great! That's what I am missing. Thank you so much @himesh-cm  for the help. It's working now..

Updated Build ID: `667971588d9f918904f3a2d0`

**Reply by himesh-cm** [reply]
Great, While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
